allennlp/nn/initializers.py

```diff
@@ -37,6 +38,50 @@ def from_params(cls, params: Params):
             return cls.by_name(choice).from_params(params)
 
 
+def block_orthogonal(tensor: torch.Tensor,
+                     split_sizes: List[int],
+                     gain: float = 1.0) -> None:
+    """
+    An initializer which allows initializing model parameters in "blocks". This is helpful
+    in the case of recurrent models which use multiple gates applied to linear projections,
+    which can be computed efficiently if they are concatenated together. However, they are
+    separate parameters which should be initialized independently.
+
+    Parameters
+    ----------
+    tensor : ``torch.Tensor``, required.
+        A tensor to initialize.
+    split_sizes : List[int], required.
+        A list of length ``tensor.ndim()`` specifying the size of the
+        blocks along that particular dimension. E.g. ``[10, 20]`` would
+        result in the tensor being split into chunks of size 10 along the
+        first dimension and 20 along the second.
+    gain : float, optional (default = 1.0)
+        The gain (scaling) applied to the orthogonal initialization.
+    """
+    sizes = list(tensor.size())
+    if any([a % b != 0 for a, b in zip(sizes, split_sizes)]):
+        raise ConfigurationError("tensor dimensions must be divisible by their respective "
+                                 "split_sizes. Found size: {} and split_sizes: {}".format(sizes, split_sizes))
+    indexes = [list(range(0, max_size, split))
+               for max_size, split in zip(sizes, split_sizes)]
+    # Our step size for each block is the split size minus 1,
+    # as the end index is exclusive.
```

Comment:
it looks like you took out the minus 1 (which is good, because I didn't get why it was there), so maybe this comment needs to be fixed?